### ISSUE TITLE
Uncaught TypeError: Cannot read property 'getSelectedText' of undefined

### ISSUE THREAD

**Issue by smrjan** [nteract/hydrogen#594]
[Enter steps to reproduce below:]

1. Ctrl + Enter a block in notebook

**Atom Version**: 1.13.0
**Electron Version**: 1.3.13
**System**: Ubuntu 16.04.1
**Thrown From**: [Hydrogen](https://github.com/nteract/hydrogen) package, v1.5.3

### Stack Trace

Uncaught TypeError: Cannot read property 'getSelectedText' of undefined

```
At /home/smrutiranjans/.atom/packages/Hydrogen/lib/code-manager.js:97

TypeError: Cannot read property 'getSelectedText' of undefined
    at CodeManager.getSelectedText (/home/smrutiranjans/.atom/packages/Hydrogen/lib/code-manager.js:97:44)
    at CodeManager.findCodeBlock (/home/smrutiranjans/.atom/packages/Hydrogen/lib/code-manager.js:15:31)
    at Object.run (/home/smrutiranjans/.atom/packages/Hydrogen/lib/main.js:358:40)
    at atom-text-editor.hydrogenRun (/home/smrutiranjans/.atom/packages/Hydrogen/lib/main.js:55:34)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app.asar/src/command-registry.js:259:29)
    at /usr/share/atom/resources/app.asar/src/command-registry.js:3:59
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:587:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:382:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/usr/share/atom/resources/app.asar/src/window-event-handler.js:106:36)
    at HTMLDocument.<anonymous> (/usr/share/atom/resources/app.asar/src/window-event-handler.js:3:59)
```

### Commands

```
     -0:01.7.0 jupyter-notebook-atom:run (input.hidden-input)
     -0:00.1.0 hydrogen:run (input.hidden-input)
```

### Config

```json
{
  "core": {
    "telemetryConsent": "limited"
  },
  "Hydrogen": {
    "kernelspec": "{\n  \"kernelspecs\": {\n    \"python2\": {\n      \"spec\": {\n        \"display_name\": \"Python 2\",\n        \"language\": \"python\",\n        \"argv\": [\"python2.7\", \"-m\", \"ipykernel\", \"-f\", \"{connection_file}\"],\n        \"env\": {}\n      }\n    },\n    \"python3\": {\n      \"spec\": {\n        \"display_name\": \"Python 3\",\n        \"language\": \"python\",\n        \"argv\": [\"python3.4\", \"-m\", \"ipykernel\", \"-f\", \"{connection_file}\"],\n        \"env\": {}\n      }\n    }\n  }\n}"
  }
}
```

### Installed Packages

```coffee
# User
atom-python-run, v0.7.3 (active)
autocomplete-python, v1.8.27 (active)
git-control, v0.8.2 (inactive)
git-plus, v7.2.0 (active)
Hydrogen, v1.5.3 (inactive)
hydrogen-launcher, v1.2.0 (active)
ink, v0.6.2 (active)
julia-client, v0.5.11 (active)
jupyter-notebook, v0.0.9 (active)
language-julia, v0.10.0 (active)
language-scala, v1.1.8 (active)
latex-completions, v0.3.2 (active)
python-tools, v0.6.9 (active)
tool-bar, v1.0.1 (active)
uber-juno, v0.1.1 (active)
atom-dark-syntax, v0.28.0 (inactive)
atom-dark-ui, v0.53.0 (inactive)
atom-light-syntax, v0.29.0 (inactive)
atom-light-ui, v0.46.0 (inactive)
base16-tomorrow-dark-theme, v1.4.0 (inactive)
base16-tomorrow-light-theme, v1.4.0 (inactive)
one-dark-ui, v1.8.2 (active)
one-light-ui, v1.8.2 (inactive)
one-dark-syntax, v1.6.0 (active)
one-light-syntax, v1.6.0 (inactive)
solarized-dark-syntax, v1.1.1 (inactive)
solarized-light-syntax, v1.1.1 (inactive)
about, v1.7.2 (active)
archive-view, v0.62.0 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.14.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.33.1 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.29.0 (inactive)
autosave, v0.23.2 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.43.2 (active)
bracket-matcher, v0.85.1 (active)
command-palette, v0.39.1 (inactive)
deprecation-cop, v0.55.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.40.0 (active)
find-and-replace, v0.204.5 (inactive)
fuzzy-finder, v1.4.0 (active)
git-diff, v1.2.0 (active)
go-to-line, v0.31.2 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.60.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.1 (active)
link, v0.31.2 (inactive)
markdown-preview, v0.159.1 (active)
metrics, v1.1.2 (active)
notifications, v0.65.1 (active)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.2 (inactive)
settings-view, v0.244.0 (active)
snippets, v1.0.4 (active)
spell-check, v0.68.5 (active)
status-bar, v1.6.0 (active)
styleguide, v0.48.0 (active)
symbols-view, v0.113.1 (inactive)
tabs, v0.103.1 (active)
timecop, v0.33.2 (active)
tree-view, v0.211.1 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.1 (active)
whitespace, v0.35.0 (active)
wrap-guide, v0.39.0 (active)
language-c, v0.54.0 (active)
language-clojure, v0.22.1 (active)
language-coffee-script, v0.48.1 (active)
language-csharp, v0.13.0 (active)
language-css, v0.40.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.43.0 (active)
language-html, v0.47.1 (active)
language-hyperlink, v0.16.1 (active)
language-java, v0.24.0 (active)
language-javascript, v0.122.0 (active)
language-json, v0.18.3 (active)
language-less, v0.29.6 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.37.0 (active)
language-php, v0.37.3 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.1 (active)
language-ruby, v0.70.2 (active)
language-ruby-on-rails, v0.25.1 (active)
language-sass, v0.57.0 (active)
language-shellscript, v0.23.0 (active)
language-source, v0.9.0 (active)
language-sql, v0.25.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.29.1 (active)
language-toml, v0.18.1 (active)
language-xml, v0.34.12 (active)
language-yaml, v0.27.1 (active)

# Dev
No dev packages
```

**Comment by lgeiger** [comment]
Hydrogen does not support running code from notebook files.

That said, we definitely shouldn't throw a exception. It looks like Hydrogen doesn't play well with [`atom-notebook`](https://github.com/jupyter/atom-notebook) and both packages try to run the same code.

I can't reproduce this at the moment because I'm running into https://github.com/atom/apm/pull/673 during the install of `atom-notebook` with Atom Beta.

**Comment by BenRussert** [comment]
We squashed this bug over the weekend, this is fixed for v1.8+
